### ISSUE TITLE
Mongo Indexes: study and add to procws and logs at least

### ISSUE THREAD

**Issue by pcampanella** [fadeoutsoftware/WASDI#451]
Indexes are a solution to speed up queries.
We should evaluate to add indexes to processworkspaces and logs at least.

**Comment by wasdi-sysadmin-br** [comment]
@pcampanella I checked quickly but I confess I'm not 100% comfortable to read the code and to understand which final queries are executed (and then which index can be nice to have).

If you have time this week or next week can we have a quick call to see together please?
I just want to see queries you have in mind: if I know queries I can create index + I can explain how it works

Just to introduce the topic "how it works".
I think (to confirm after a more complete analysis) we will work with 2 types of index:
- an index on a single field: example index on 'firstname'
- a compound index on several fields:
    - example -> index = firstname + lastname:
      - in this case, we have 2 indexes:
        - firstname + lastname
        - firstname
      - the order is important so we don't have (it means we have no index so we will do a collection scan):
        - lastname + firstname
        - lastname

    - example -> index = firstname + lastname + city:
      - in this case, we have 3 indexes:
        - firstname + lastname + city
        - firstname + lastname
        - firstname
      - the order is important so we don't have (it means we have no index so we will do a collection scan):
        - firstname + city + lastname
        - lastname + firstname
        - etc

It can be nice to work with compound index but we have to be sure queries use always the same order for field :)

Example:
- we have a query 1 with a filter on firstname = "value1" and lastname = "value2"
- we have a query 2 with a filter on lastname = "value2" and firstname = "value1" and city ="value3"

In this case, it can be nice to have a compound index on firstname + lastname + city
And to change query 2:
- before: query 2 with a filter on lastname = "value2" and firstname = "value1" and city ="value3"
- after: query 2 with a filter on firstname = "value1" and lastname = "value2" and city ="value3"

It's not necessary to have 2 indexes like:
  - index 1 = a compound index on firstname + lastname
  - index 2 = a single index on firstname because firstname will be used from the index 1
  
But it can be necessary (it depends of usecase, etc) to have:
  - index 1 = a compound index on firstname + lastname
  - index 2 = a single index on lastname because lastname can't be used from the index 1

**Comment by pcampanella** [comment]
We need to add:
ProcessWorkspace:
 .one index on processObjId
 .one compund index on status + nodeCode
 
ProcessorsLog:
 .one index on processWorkspaceId
 .one compund index  processWorkspaceId + rowNumber

**Comment by wasdi-sysadmin-br** [comment]
Thank you for the update :) 

Just quick questions:
- is processObjId must be unique?
- is processWorkspaceId must be unique?

For processWorkspaceId + rowNumber I have a doubt and will send you a message on slack to be sure about something

**Comment by wasdi-sysadmin-br** [comment]
**Update**

```
For processWorkspaceId + rowNumber I have a doubt and will send you a message on slack to be sure about something
```

Paolo gave we the answer on slack and it is really a query on processWorkspaceId  AND rowNumber .

-----

I made tests and for me only these indexes are necessary:
* processworkpsace > processObjId
* processworkpsace > status
* processorlog > processWorkspaceId

**Comment by wasdi-sysadmin-br** [comment]
**Update**

Indexes are created in TEST only.

**Comment by wasdi-sysadmin-br** [comment]
**Update**

I made a test with a query considered as "slow" in logs: now this query use the indexes I created.
For me these indexes can be deployed everywhere.

**Comment by wasdi-sysadmin-br** [comment]
**Update**

Indexes was deployed on the 2022-02-09 on all servers

**Comment by wasdi-sysadmin-br** [comment]
**Update**

I detected a new missing index on processworkpsace.productName.
I just deployed an index in TEST.

Then to resume, we have 2 indexes in TEST not deployed in PROD:

- counter.sequence
- processworkpsace.productName

**Comment by wasdi-sysadmin-br** [comment]
**Update**

Because of my ongoing tasks with JMeter I detected new missing indexes.
Then at this moment we have these indexes in TEST not deployed in PROD:

- counter: sequence
- processorsharing: userId + processorId
- processworkpsace: productName
- reviews: processorId
- sessions: sessionId
- sessions: lastTouch
- users: userId
- workspaces: workspaceId
- workspacessharing: workspaceId
